source/state_representation/test/tests/robot/test_joint_accelerations.cpp

```diff
@@ -1,79 +1,187 @@
 #include <gtest/gtest.h>
 #include "state_representation/robot/JointAccelerations.hpp"
+#include "state_representation/exceptions/EmptyStateException.hpp"
 
 using namespace state_representation;
 
-TEST(JointAccelerationsTest, RandomAccelerationsInitialization) {
+TEST(JointAccelerationsTest, Constructors) {
+  std::vector<std::string> joint_names{"joint_10", "joint_20"};
+  Eigen::Vector2d accelerations = Eigen::Vector2d::Random();
+  JointAccelerations ja1("test", accelerations);
+  EXPECT_EQ(ja1.get_name(), "test");
+  EXPECT_FALSE(ja1.is_empty());
+  EXPECT_EQ(ja1.get_size(), accelerations.size());
+  for (std::size_t i = 0; i < accelerations.size(); ++i) {
+    EXPECT_EQ(ja1.get_names().at(i), "joint" + std::to_string(i));
+  }
+  EXPECT_EQ(ja1.data(), accelerations);
+
+  JointAccelerations ja2("test", joint_names, accelerations);
+  EXPECT_EQ(ja2.get_name(), "test");
+  EXPECT_FALSE(ja2.is_empty());
+  EXPECT_EQ(ja2.get_size(), joint_names.size());
+  for (std::size_t i = 0; i < joint_names.size(); ++i) {
+    EXPECT_EQ(ja2.get_names().at(i), joint_names.at(i));
+  }
+  EXPECT_EQ(ja2.data(), accelerations);
+}
+
+TEST(JointAccelerationsTest, StateCopyConstructor) {
+  JointState random_state = JointState::Random("test", 3);
+  JointAccelerations copy1(random_state);
+  EXPECT_EQ(random_state.get_name(), copy1.get_name());
+  EXPECT_EQ(random_state.get_names(), copy1.get_names());
+  EXPECT_EQ(random_state.get_size(), copy1.get_size());
+  EXPECT_EQ(random_state.get_accelerations(), copy1.data());
+  EXPECT_EQ(static_cast<JointState&>(copy1).get_positions().norm(), 0);
+  EXPECT_EQ(static_cast<JointState&>(copy1).get_velocities().norm(), 0);
+  EXPECT_EQ(static_cast<JointState&>(copy1).get_torques().norm(), 0);
```

Comment:
There's some opportunities in general to make these tests less verbose by putting common expectations in static test methods. This block in particular comes up 4 times, with a similar thing in the other files. 

You can instead stick this function up top and reuse it four times. (static keyword is there just for safety, to limit it to just this source file)

```c++
static void expect_only_acceleration(const JointAccelerations& acc) {
  EXPECT_EQ(static_cast<JointState&>(copy1).get_positions().norm(), 0);
  EXPECT_EQ(static_cast<JointState&>(copy1).get_velocities().norm(), 0);
  EXPECT_EQ(static_cast<JointState&>(copy1).get_torques().norm(), 0);
}
```

Reply:
This is actually super nice. Thanks for pointing out this pattern